user/views.py

```diff
@@ -0,0 +1,58 @@
+from rest_framework.views import APIView
+from rest_framework import authentication
+from rest_framework.response import Response
+from rest_framework import status
+from .serializers import UserSerializer
+from .models import User
+from django.contrib.auth import authenticate
+from rest_framework.authtoken.models import Token
+ # Create your views here.
+
+class Userview(APIView):
+    authentication_classes = [authentication.TokenAuthentication]
+    serializer_class = UserSerializer
+
+    
+    def get(self, request, id=None):
+        user = User.objects.all()
+        serializer = self.serializer_class(user, many=True)
+        return Response(serializer.data)
+                    
+    
+    def post(self, request):
+        user = User.objects.create_user(
+            name = request.data.get('name'),
+            email = request.data.get('email'),
+            password = request.data.get('password'),
+        )
+        serializer =  UserSerializer(user)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+    
+    
+    
+    def patch(self, request, id=None):
+        try:
+            user = User.objects.get(id=id)
+            serializer = self.serializer_class(user, data=request.data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+            return Response(serializer.data)
```

Comment:
what will happen if serializer not valid?

Reply:
fixed